lib/std/fs.zig

```diff
@@ -574,10 +593,14 @@ pub const Dir = struct {
             pub fn next(self: *Self) Error!?Entry {
                 start_over: while (true) {
                     if (self.index >= self.end_index) {
+                        if (self.first_iter) {
+                            std.os.lseek_SET(self.dir.fd, 0) catch unreachable; // Can't fail
+                            self.first_iter = false;
+                        }
                         const rc = linux.getdents64(self.dir.fd, &self.buf, self.buf.len);
                         switch (linux.getErrno(rc)) {
                             .SUCCESS => {},
-                            .BADF => unreachable, // Dir is invalid or was opened without iteration ability
+                            .BADF => unreachable, // EBADF here likely means that the Dir was not opened with iteration permissions
```

Comment:
This comment can probably stay as it was before.

Reply:
Ah I confused the two lines ;)